### PR TITLE
IDE-2716 change detection of liferayHome and create modules dir if no…

### PR DESCRIPTION
**Pull request by AndyWu2015**
…t existing

### CODE REVIEW

**Comment by yanan** [comment]
Please help merge this, then we can add portal-ce-ga2 server on our ide. It build succesfully.

File: tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundleFactory.java [hunk 1]
```diff
@@ -112,9 +112,8 @@ private boolean detectLiferayHome( IPath path )
             return false;
         }
 
-        if( path.append( "osgi/core" ).toFile().exists() &&
-            path.append( "osgi/modules" ).toFile().exists() &&
-            path.append( "osgi/portal" ).toFile().exists() )
+        if( path.append( "data" ).toFile().exists() &&
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundleFactory.java]: Data doesn't necessaryily have to exist.  Also previously the data folder didn't have to exist in GA1 we need this to support both GA1 and GA2

**Comment by gamerson** [comment]
We need to make sure this code works for both GA1 and GA2  can you test it?

On Mon, Jun 13, 2016 at 12:23 AM, Ashley Yuan <notifications@github.com>
wrote:

> Please help merge this, then we can add portal-ce-ga2 server on our ide.
> It build succesfully.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gamerson/liferay-ide/pull/1590#issuecomment-225491537>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAkVFf34os4cE6bayRg2OmmkBLYg1kvBks5qLOlYgaJpZM4Iz9wt>
> .
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

File: tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundleFactory.java (hunk 1 above)
Comment (AndyWu2015) [inline tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundleFactory.java]: I've tested it.
maybe only osgi and license have to exist.

Comment (gamerson) [inline tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundleFactory.java]: Lets just do the osgi folder, that hsould be ok i think

**Comment by gamerson** [comment]
Try to resend before 2 hours when I have to leave for my flight. thanks! @yanan @AndyWu2015

**Comment by AndyWu2015** [comment]
sure , two minutes

**Pull request closed without merging**